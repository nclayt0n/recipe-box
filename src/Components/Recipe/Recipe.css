/* * {
    box-sizing: border-box;
}

.recipe {
    margin: 10px 15px 15px 75px;
    display: flex;
    flex-direction: column;
    width: 70%;
}

.recipeItems {
    background-image: url('../../images/lavleaves.png');
    background-repeat: no-repeat;
    background-size: cover;
    background-color: white;
    padding: 15px;
    margin: 10px;
}

h3 {
    font-size: 26px;
    margin: 10px;
}

.recipe ul {
    text-align: center;
    margin: 5px;
    padding: 5px;
    background-color: rgba(255, 255, 255, 0.562);
    border: 1px solid black;
}

.recipe li {
    letter-spacing: 2px;
    margin: auto;
    padding: 10px;
    text-align: left;
    list-style: none;
    /* text-decoration: underline; */


}
*/