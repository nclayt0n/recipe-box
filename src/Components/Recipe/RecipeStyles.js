import React from 'react'
import navImage from '../../images/lavleaves.png'
const recipeStyles = {
    mobile: {
        recipe: {
            margin: '10px 15px 15px 75px',
            display: 'flex',
            flexDirection: 'column',
            width: '70%',
            fontSize:'4vw'
        },

        recipeItems: {
            backgroundImage: `url(${navImage})`,
            backgroundRepeat: 'noRepeat',
            backgroundSize: 'cover',
            backgroundColor: 'white',
            padding: '15px',
            margin: '10px'
        },

        h3: {
            margin: 'auto',
            backgroundColor:'var(--purple)',
            color:'white',
            width:'fit-content',
            textShadow:'1px 1px rgba(138, 100, 163, 0.733)',
        },
        recipeUl: {
            textAlign: 'center',
            margin: '5px',
            padding: '15px',
            // border: '1px solid black',
            fontWeight:'bolder',
            color:'#5E1770'
        },

        recipeLi:{
            letterSpacing: '2 px',
            margin: 'auto',
            padding: '15 px',
            textAlign: 'center',
            listStyle: 'none',
            fontWeight:'normal',
            color:'var(--purple)'
            /* text-decoration: underline; */
        },
        buttons:{
            display:'flex',
            width:'fit-content',
            margin:'auto',
            flexDirection:'column'
        },
        recipeButton:{
            width:'fit-content',
            margin:'auto',
            marginBottom:'5px',
            fontSize:'3vw',
            border:'2px solid var(--purple)',
            backgroundColor:'var(--purple)',
            padding:'5px',
            fontWeight:'bolder',
            color:'white'
        },
        folderButton:{
            width:'fit-content',
            margin:'auto'
            
        }
    },
    laptop: {
        recipe: {
            margin: 'auto',
            display: 'flex',
            flexDirection: 'column',
            width: '60%',
            padding:'10px',
            fontSize:'2vw',
            
        },

        recipeItems: {
            backgroundImage: `url(${navImage})`,
            backgroundRepeat: 'noRepeat',
            backgroundSize: 'cover',
            backgroundColor: 'white',
            padding: '15px',
            margin: '10px',
            
        },

        h3: {
            fontSize:'2.5vw',
            margin: 'auto auto 20px',
            backgroundColor:'var(--purple)',
            color:'white',
            width:'fit-content',
            textShadow:'1px 1px rgba(138, 100, 163, 0.733)',
        },
        recipeUl: {
            textAlign: 'center',
            width:'60%',
            margin: 'auto auto 15px',
            padding: '15px',
            backgroundColor: 'rgba(255, 255, 255, 0.9)',
            fontWeight:'bolder',
            color:'var(--purple)',
            border:'1px solid var(--purple)',
            textShadow:'.5px .5px rgba(138, 100, 163, 0.733)',
        },

        recipeLi:{
            letterSpacing: '2 px',
            width:'60%',
            margin: 'auto',
            padding: '10 px',
            textAlign: 'center',
            listStyle: 'none',
            textDecoration: 'none',
            fontWeight:'normal',
            color:'var(--purple)'
        },
        buttons:{
            display:'flex',
            width:'fit-content',
            margin:'auto',
            
        },
        recipeButton:{
            width:'fit-content',
            margin:'5px',
            fontSize:'1.5vw',
            backgroundColor:'var(--purple)',
            padding:'5px',
            fontWeight:'bolder',
            color:'white'
        },
        folderButton:{
            width:'fit-content',

        }
    }

}
export default recipeStyles