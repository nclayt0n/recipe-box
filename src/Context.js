import React from 'react'

const Context = React.createContext({
    folders: [],
    recipes: [],
    user_id: '',
    addRecipe: () => {},
    addFolder: () => {},
    addFolders: () => {},
    addRecipes: () => {},
    updateRecipe: () => {},
    updateFolder: () => {},
    deleteRecipe: () => {},
    deleteFolder: () => {},

})
export default Context