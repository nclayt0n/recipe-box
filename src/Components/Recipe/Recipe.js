import React from 'react'
import {withRouter,Link} from 'react-router-dom'
import Context from '../../Context'
const uuidv4 = require('uuid/v4');


function findFolderandRecipe(id,folders,recipes){
const results=recipes.filter(recipe=>recipe.id===id);
const recipe=results[0];
const folder = folders.filter(f =>f.id ===recipe.folderId );
recipe.folderName=folder[0].name;
return recipe;
}

class Recipe extends React.Component{
    static contextType=Context;
    createDisplayedIngredients=(ingredients)=>{
        return ingredients.map(ingredient=>{return<li key={uuidv4()}>{ingredient.name} {ingredient.quantity} {ingredient.unit}</li>})}
    render(){
        console.log(this.props)
        let recipe=findFolderandRecipe(this.props.match.params.id,this.context.folders,this.context.recipes);
        const ingredients=this.createDisplayedIngredients(recipe.ingredients)
        //will need to add measurements and break apart into ingrdients
        return(
            <div className='recipe'>
            <h3>Recipe</h3>
                <ul>{recipe.name}
                <li>INTRUCTIONS: {recipe.instructions}</li>
                <Link to={'/ingredients'}>INGREDIENTS:</Link> 
               {ingredients}
                <li>Created By: {recipe.createdBy}</li>
                <li>{recipe.link}</li>
                <button><Link to={`/folder/${recipe.folderId}`}>{recipe.folderName}</Link></button>
                </ul>
                <button><Link to={`/update-recipe/${recipe.id}`}>Update Recipe</Link></button>
                <button onClick={()=>this.props.history.goBack()}>Back</button><br/>
                <button type='button' onClick={()=>this.context.deleteRecipe(recipe.id,this.props)}>Delete Recipe</button>
            </div>
        )
    }
}
export default withRouter(Recipe)