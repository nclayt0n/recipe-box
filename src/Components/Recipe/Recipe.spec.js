import React from 'react'
import ReactDOM from 'react-dom';
import Recipe from './Recipe'
import { shallow } from 'enzyme'
import {BrowserRouter} from 'react-router-dom'
import toJson from 'enzyme-to-json'
import * as Context from '../../Context'
 import ShallowRenderer from 'react-test-renderer/shallow';
 
 describe.skip('<Recipe />', () => {
     const contextValues = { folders: [{
        "id": "d26e01a6-ffaf-11e8-8eb2-f2801f1b9fc3",
        "name": "Spangley",
        "user_id": 1
     },{
        "id": "d26e01a6-ffaf-11e8-8eb2-f2801f1b9fc3",
        "name": "Spangley",
        "user_id": 1
     }]}
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<BrowserRouter><Recipe/></BrowserRouter>,div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('snapshot', () => {
const wrapper=shallow(<Recipe contextValues={contextValues}/>)
expect(toJson(wrapper))
.toMatchSnapshot()
     console.log(Context)
});
     })
