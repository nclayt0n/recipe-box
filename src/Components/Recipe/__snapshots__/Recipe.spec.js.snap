// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Recipe /> snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
